security/subsystem/src/main/java/org/jboss/as/security/SecuritySubsystemParser.java

```diff
@@ -158,14 +162,20 @@ public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) th
                     parseProperties(Element.VAULT_OPTION.getLocalName(), reader, vault, VaultResourceDefinition.OPTIONS);
                     break;
                 }
+                case ELYTRON_INTEGRATION: {
+                    // the elytron-integration element was added in version 2.0 of the subsystem.
+                    if (schemaVer.compareTo(Namespace.SECURITY_2_0) < 0) {
```

Comment:
TBH I prefer parsers to just fork on a new major version - it does mean duplication of code but it also simplifies the comparisons we end up with trying to compare versions.

Having said that we have had discussions in the app server team and silently supporting an optional new element or attribute is not an issue.